packages/core/feature-flags/README.mdx

```diff
@@ -0,0 +1,158 @@
+
+## Registering a feature flag
+
+Kibana follows a _gitops_ approach when managing feature flags. To declare a feature flag, add your flags definitions in
+your plugin's `server/index.ts` file:
+
+```typescript
+// <plugin>/server/index.ts
+import type { FeatureFlagDefinitions } from '@kbn/core-feature-flags-server';
+import type { PluginInitializerContext } from '@kbn/core-plugins-server';
+
+export const featureFlags: FeatureFlagDefinitions = [
+  {
+    key: 'my-cool-feature',
+    name: 'My cool feature',
+    description: 'Enables the cool feature to auto-hide the navigation bar',
+    tags: ['my-plugin', 'my-service', 'ui'],
+    variationType: 'boolean',
+    variations: [
+      {
+        name: 'On',
+        description: 'Auto-hides the bar',
+        value: true,
+      },
+      {
+        name: 'Off',
+        description: 'Static always-on',
+        value: false,
+      },
+    ],
+  },
+  {...},
+];
+
+export async function plugin(initializerContext: PluginInitializerContext) {
+  const { FeatureFlagsExamplePlugin } = await import('./plugin');
+  return new FeatureFlagsExamplePlugin(initializerContext);
+}
+```
+
+After merging your PR, the CI will create/update the flags in our third-party feature flags provider.
+
+### Deprecation/removal strategy
+
+When your code doesn't use the feature flag anymore, it is recommended to clean up the feature flags when possible.
+There are a few considerations to take into account when performing this clean-up:
+
+1. Always deprecate first, remove after
+2. When to remove?
+
+#### Always deprecate first, remove after
+
+Just because the CI syncs the state of `main` to our feature flag provider, there is a high probability that the
+previous version of the code that still relied on the feature flag is still running out there.
+
+For that reason, the recommendation is to always deprecate before removing the flags. This will keep evaluating the flags,
+according to the segmentation rules configured for the flag.
+
+#### When to remove?
+
+After deprecation, we need to consider when it's safe to remove the flag. There are different scenarios that come with
+different recommendations:
+
+* The segmentation rules of my flag are set up to return the fallback value 100% of the time: it should be safe to
+remove the flag at any time.
+* My flag only made it to Serverless (it never made it to Elastic Cloud Hosted): it should be safe to remove the flag
+after 2 releases have been rolled out (roughly 2-3 weeks later). This is to ensure that all Serverless projects have 
+been upgraded and that we won't need to rollback to the previous version.
+* My flag made it to Elastic Cloud Hosted: if we want to remove the flag, we should approach the affected customers to
+fix the expected values via [config overrides](#config-overrides).
+
+In general, the recommendation is to check our telemetry to validate the usage of our flags.
```

Comment:
Not implemented in this PR because it requires prior work on core (UI counters are not available to core services yet 😬), but the intention is to increase UI/Usage counters to count the number of evaluations of a flag and the returned value.

For now, in this PR, APM labels will provide the insights.